File: crates/cloudpipe-sdk/examples/quickstart.rs
//! Minimal quickstart — start a tunnel to `localhost:8080` and print its URL.
//!
//! Run with:
//!
//! ```text
//! CLOUDFLARE_API_TOKEN=... \
//! CFP_DOMAIN=example.com \
//! cargo run --example quickstart
//! ```
//!
//! Account ID and zone ID are looked up automatically by the SDK.

use cloudpipe_sdk::{Protocol, TunnelBuilder};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let token = std::env::var("CLOUDFLARE_API_TOKEN")?;
    let domain = std::env::var("CFP_DOMAIN")?;

    let handle = TunnelBuilder::new()
        .token(token)
        .domain(domain)
        .protocol(Protocol::Http)
        .port(8080)
        .subdomain("quickstart")
        .start()
        .await?;

    println!("live at {}", handle.url());
    handle.wait().await;
    Ok(())
}

File: crates/cloudpipe-node/examples/quickstart.js
'use strict';

// Minimal quickstart — expose localhost:8080 to the public internet.
//
// Prereqs:
//   - CLOUDFLARE_API_TOKEN env var
//   - CFP_DOMAIN env var (a zone this token can write DNS for)
//
// Run with:
//   node examples/quickstart.js

const { connect } = require('cloudpipe');

(async () => {
 const token = process.env.CLOUDFLARE_API_TOKEN;
 const domain = process.env.CFP_DOMAIN;
 if (!token || !domain) {
 console.error(
 'set CLOUDFLARE_API_TOKEN and CFP_DOMAIN before running this example',
 );
 process.exit(1);
 }

 const listener = await connect({
 token,
 domain,
 protocol: 'http',
 port: 8080,
 });

 console.log('live at', listener.url);
 console.log('subdomain:', listener.subdomain);

 process.on('SIGINT', async () => {
 console.log('shutting down...');
 try {
 await listener.close();
 } catch (err) {
 console.error('shutdown error:', err);
 }
 process.exit(0);
 });

 // Wait for ctrl-c; the listener emits lifecycle events on the same
 // object if you also want to react to cloudflared crashes.
 await listener.wait().catch((err) => {
 console.error('tunnel exited unexpectedly:', err);
 });
})();